### PR TITLE
Add attachment upload support to SDK

### PR DESCRIPTION
**Pull request by johnytex**
This PR adds support for uploading attachments through the SDK.

### CODE REVIEW

File: allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropAPI.java [hunk 1]
```diff
@@ -31,7 +33,7 @@
 
 public class DropAPI {
 
-   public static List<String> getUrls(int count) throws GraphQLException {
+    public static List<String> getUrls(int count) throws GraphQLException {
```
Comment (robindowling) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropAPI.java]: ⁉️

Comment (johnytex) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropAPI.java]: These lines were not indented in a way that Android Studio agrees with, so the `Reformat Code` command changed these lines.

File: allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropPiece.java [hunk 2]
```diff
@@ -166,6 +182,58 @@ JSONObject toJSON(boolean validate) throws JSONException {
...
+         * Set the public attachment URL. Call this method after an attachment URL has been generated
+         * and an attachment succesfully uploaded.
+         * @param url Required The final URL of the uploaded attachment
+         */
+        public void setAttachmentURL(@NonNull String url) {
+            this.attachmentURL = url;
+        }
+
+        private void checkInvariant() {
+            Objects.requireNonNull(fileResource, "fileResource must not be null");
```
Comment (tomhai) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropPiece.java]: Nice 👍

File: allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropTask.java [hunk 3]
```diff
@@ -7,22 +7,23 @@
...
 import java.io.IOException;
+import java.net.URL;
 import java.util.Iterator;
 import java.util.List;
 
 /**
  * DropTask takes care of dropping a piece to Allihoopa. This task is executed as a result of
  * user clicking a drop button on {@link com.allihoopa.sdk.drop.fragments.PieceFragment}.
- *
+ * <p>
```
Comment (tomhai) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropTask.java]: Here too it looks like it is missing the closing `</p>` tags.

File: allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/FileResource.java [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 /**
  * Wrapper for the file type and file input stream for passing files to the AllihoopaSDK.
- *
+ * <p>
```
Comment (tomhai) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/FileResource.java]: Same `</p>`

Comment (johnytex) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/FileResource.java]: Android Studio keeps inserting these when I do `Reformat Code` - I don't know why it doesn't insert closing `</p>` tags

File: allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropTask.java [hunk 5]
```diff
@@ -56,19 +57,31 @@ public DropTask(DropPiece dropPiece, DropTaskCallback listener) {
...
+            if (mDropPiece.getPresentation().hasCoverImageFileResource()) {
                 String coverImageURL = urlIterator.next();
                 DropAPI.uploadFile(coverImageURL, mDropPiece.getPresentation().getCoverImageFileResource().getFileInputStream());
                 mDropPiece.getPresentation().getCoverImage().setFileURL(coverImageURL);
             }
 
+            if (mDropPiece.hasAttachment()) {
+                DropPiece.Attachment attachment = mDropPiece.getAttachment();
+                String mimeType = attachment.getMimeType();
+                String url = DropAPI.getUploadURLForAttachment(mimeType).toString();
```
Comment (johnytex) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropTask.java]: I would prefer changing the rest of the API to use the `URL` type - it adds type safety and indicates to the user that it's a valid URL.

Comment (tomhai) [inline allihoopa-sdk/src/main/java/com/allihoopa/sdk/drop/DropTask.java]: That would be an ideal solution 👍

**Pull request merged**